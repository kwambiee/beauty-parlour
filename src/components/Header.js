import React from 'react';

const Header = () => (
  <div className="header">
    <h1>
      DROP-DEAD
      {' '}
      <br />
      GORGEOUS
    </h1>
    <button type="button" className="button">
      SHOP NOW
    </button>
  </div>
);
export default Header;
