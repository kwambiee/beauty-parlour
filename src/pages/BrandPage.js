import React from "react";
import { useParams, Outlet } from "react-router-dom";

import { useSelector } from "react-redux";
import Product from "../components/product";

const BrandPage = () => {
  const params = useParams();
  const brandNames = useSelector((state) => state.brands.brands);
  const products = brandNames[params.brandName];
  console.log(products);

  return (
    <div>
      {products.map((product) => {
        <Product products={product} />;
      })}
      <Outlet />
    </div>
  );
};
export default BrandPage;
