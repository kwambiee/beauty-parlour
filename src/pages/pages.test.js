import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import BrandPage from "./BrandPage";
import HomePage from "./Homepage";
import Productpage from "./ProductPage";
import Store from "../Redux/ConfigureStore";
import Header from "../components/Header";

const brandPage = () => (
  <Provider store={Store}>
    <BrandPage />
  </Provider>
);
const homePage = () => (
  <Provider store={Store}>
    <HomePage />
  </Provider>
);

const prodPage = () => (
  <Provider store={Store}>
    <Productpage />
  </Provider>
);

const MockPage = () => (
  <Provider store={Store}>
    <Header />
  </Provider>
);

describe("Check Brand Page", () => {
  it("renders correctly", () => {
    const component = renderer.create(<brandPage />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe("Check Home Page", () => {
  it("renders correctly", () => {
    const component = renderer.create(<homePage />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe("Check Home Page", () => {
  it("renders correctly", () => {
    const component = renderer.create(<prodPage />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
