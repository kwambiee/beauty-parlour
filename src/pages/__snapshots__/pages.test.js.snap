// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Brand Page renders correctly 1`] = `<brandPage />`;

exports[`Check Home Page renders correctly 1`] = `<homePage />`;

exports[`Check Home Page renders correctly 2`] = `<prodPage />`;
